src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsUserManagementService.java

```diff
@@ -159,69 +161,74 @@ public void deleteUser(User user) throws ContinuousIntegrationException {
      * the user if it doesn't exist.
      * <p>
      * Note that it's not possible to change the username of the Jenkins user.
+     * We can't support updating the user without setting a new password at the moment. See {@link #updateUserLogin(String, User, String)} for more information
      *
-     * @param user The user to update.
+     * @param user     The user to update.
+     * @param password The user's password
      */
     @Override
-    public void updateUser(User user) throws ContinuousIntegrationException {
+    public void updateUser(User user, String password) throws ContinuousIntegrationException {
         if (user.getLogin().equals(jenkinsAdminUsername)) {
             log.debug("Jenkins updateUser: Skipping jenkinsAdminUser: {}.", jenkinsAdminUsername);
             return;
         }
+        if (password == null) {
+            return;
+        }
 
         // Only update a user if it exists.
         var jenkinsUser = getUser(user.getLogin());
         if (jenkinsUser != null) {
             deleteUser(user);
-            createUser(user);
+            createUser(user, password);
         }
     }
 
     /**
-     * Updates the user login of the Jenkins user. Jenkins uses the user login as the unique
-     * id. In order to change that, we delete the Jenkins user with the old login and create
-     * a new one with the new login.
+     * We don't support this for Jenkins because Jenkins doesn't support such a user update
+     * We would have to delete and recreate the user, but we can't do this without a new password
+     * In the future this could be supported by forcing to set a new password on update.
      *
      * @param oldLogin the old login
-     * @param user The Artemis user with the new login
+     * @param user     The Artemis user with the new login
+     * @param password The user's password
      * @throws ContinuousIntegrationException if something went wrong deleting/creating the user
      */
     @Override
-    public void updateUserLogin(String oldLogin, User user) throws ContinuousIntegrationException {
-        if (oldLogin.equals(user.getLogin())) {
-            return;
-        }
-
-        // We create a new user object which has the old user login and groups.
-        // We do this to revoke the old permissions for that user before creating
-        // the new one.
-        var oldUser = new User();
-        oldUser.setLogin(oldLogin);
-        oldUser.setGroups(getUserWithGroups(user).getGroups());
-        deleteUser(oldUser);
-
-        createUser(user);
+    public void updateUserLogin(String oldLogin, User user, String password) throws ContinuousIntegrationException {
     }
```

Comment:
```suggestion
    public void updateUserLogin(String oldLogin, User user, String password) throws ContinuousIntegrationException {
        // Intentionally empty. See JavaDoc for details.
    }
```
Codacy wants the empty method to be documented.